src/main/java/ch/wisv/areafiftylan/service/TaskScheduler.java

```diff
@@ -44,8 +58,20 @@ public void ExpireOrders() {
         expiredOrders.forEach(o -> orderService.expireOrder(o));
     }
 
+    @Scheduled(fixedRate = USER_CLEANUP_CHECK_INTERVAL_MINUTES * 60 * 1000)
+    public void CleanUpUsers() {
+        LocalDateTime now = LocalDateTime.now();
+        List<VerificationToken> allExpiredVerificationTokens = verificationTokenRepository.findAllByExpiryDateBefore(now);
+        allExpiredVerificationTokens.forEach(t -> handleExpiredVerificationToken(t));
+    }
+
     public static Predicate<Order> isExpired() {
         return o -> o.getStatus().equals(OrderStatus.CREATING) || o.getStatus().equals(OrderStatus.EXPIRED) ||
                 o.getStatus().equals(OrderStatus.CANCELLED);
     }
+
+    private void handleExpiredVerificationToken(VerificationToken verificationToken) {
+        verificationTokenRepository.delete(verificationToken);
+        userRepository.delete(verificationToken.getUser());
```

Comment:
A simple manual invoke could test this function. This Scheduler can be autowired into the test, which could allow for this.

Also. There's no such thing as "untestable". Adten :stuck_out_tongue: